### PR TITLE
Also generate metrics for training and validation data

### PR DESCRIPTION
**Pull request by xuhdev**
This helps hyperparamter tuning by comparing the metrics on training
data and test data.

### CODE REVIEW

File: training/training_code/train_ner.py [hunk 1]
```diff
@@ -628,6 +628,15 @@ def ftags(name):
     # train confidence branch
     ner_model = train_confidence_branch(ner_model, params, train_data, valid_data, ckpt_dir)
 
+    # generate entity-level metrics for training data
```
Comment (MLnick) [inline training/training_code/train_ner.py:631]: Shall we just compute for `train`, `validation` and `test` then?

i.e. `for dataset in ['train', 'valid', 'test']:` ?

Comment (MLnick) [reply training/training_code/train_ner.py:631]: I also need to update the metrics computation to be entity-level rather than tag-level

Comment (xuhdev) [reply training/training_code/train_ner.py:631]: I think test data is usually inaccessible to the user during the development, and should only be used in evaluation? (I believe the `test_` variables in this script actually refer to validation?)

Comment (MLnick) [reply training/training_code/train_ner.py:631]: One would expect anyone training to have a train, valid and test set. Here it refer to test set.

Hence why if we will compute metrics for training, we may as well compute for all 3 sets

Comment (xuhdev) [reply training/training_code/train_ner.py:631]: Got it, thought on something different. Let me change the script

File: training/training_code/train_ner.py [hunk 2]
```diff
@@ -628,6 +628,24 @@ def ftags(name):
...
+
+    # generate entity-level metrics for validation data
+    print('Computing multi-class metrics for validation data')
+    valid_scores = generate_mc_metrics(params, ner_model, valid_data)
+    print('Validation scores:')
+    print(valid_scores)
+    csv_out = str(Path(RESULT_DIR, 'valid_scores.csv'))
+    valid_scores.to_csv(csv_out, index=False)
+    print('Wrote scores to: {}'.format(csv_out))
+
```
Comment (MLnick) [inline training/training_code/train_ner.py:648]: @xuhdev can we extract this into a for loop? Since we're doing it 3 times over it's repeating quite a bit of code.
